lib/puppet/network/authconfig.rb

```diff
@@ -89,4 +89,28 @@ def initialize(rights=nil)
       insert_default_acl
     end
   end
+
+  class Network::AuthConfig
+    @@authprovider_class = nil
+
+    def self.authprovider_class=(klass)
+      @@authprovider_class = klass
+    end
+
+    def self.authprovider_class
+      @@authprovider_class || Puppet::Network::DefaultAuthProvider
+    end
+
+    def initialize(rights=nil)
+      if rights.nil?
```

Comment:
We need the if because `someclass.new(nil)` is not the same as `someclass.new()`.

This way it allows for the authprovider class to use a default value for the initialize parameter.
Without the if we'd just be passing in nil and not allowing the class to use their own default.